vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go

```diff
@@ -28,7 +28,7 @@ import (
 const LoopbackClientServerNameOverride = "apiserver-loopback-client"
 
 func (s *SecureServingInfo) NewClientConfig(caCert []byte) (*restclient.Config, error) {
-	if s == nil || (s.Cert == nil && len(s.SNICerts) == 0) {
+	if s == nil || (s.DynamicCertificates == nil) {
```

Comment:
why is DynamicCertificates a pointer if it is linked to tightly to SecureServingInfo?